service/src/client.rs

```diff
@@ -0,0 +1,198 @@
+//! Post Service GRPC client
+//!
+//! This module implements a GRPC client for the Post Service.
+//! It connects to the node and registers itself as a Post Service.
+//! It then waits for requests from the node and forwards them to the Post Service.
+
+use std::time::Duration;
+
+pub(crate) use spacemesh_v1::post_service_client::PostServiceClient;
+use spacemesh_v1::{node_request, service_response};
+use spacemesh_v1::{
+    GenProofRequest, GenProofResponse, GenProofStatus, Proof, ProofMetadata, ServiceResponse,
+};
+use tokio::sync::mpsc;
+use tokio::time::sleep;
```

Comment:
i think implementation will be simpler with 2 threads that communicate over a single subscriber channel. one thread connects to node and dispatches commands from node, and another writes responses.

or it is not worth it and using tokio (other event loop) is just a good default option?

Reply:
I'm not entirely sure what you mean exactly. Do you mean to have an actor-like pattern where the client and service run in a loop and communicate via a channel instead of just directly calling service methods from the client? It is possible, but IMHO will complicate the code as the calls would become "async" (send on a channel and await response on another channel).